### ISSUE TITLE
Testing model performance is taking too long...

### ISSUE THREAD

**Issue by eashan2798** [TensorSpeech/TensorFlowASR#173]
When testing the performance of the model it is taking almost 2 hour for only 6 instances(rows).
Ideally It should generate the results within the span of minutes.

```
python test_subword_conformer.py \
--config 'config.yml' \
--subwords '/Model/subword-conformer/conformer.subwords' \
--saved '/Model/subword-conformer/latest.h5' \
--output_name 'test_results'
```

**Comment by usimarit** [comment]
@eashan2798 Can you show me the config.yml?

**Comment by eashan2798** [comment]
speech_config:
  sample_rate: 16000
  frame_ms: 25
  stride_ms: 10
  num_feature_bins: 80
  feature_type: log_mel_spectrogram
  preemphasis: 0.97
  normalize_signal: True
  normalize_feature: True
  normalize_per_feature: False

decoder_config:
  vocabulary: null
  target_vocab_size: 4096
  max_subword_length: 4
  blank_at_zero: True
  beam_width: 5
  norm_score: True
  corpus_files:
    - '/content/drive/MyDrive/Colab Notebooks/TestingAudios/training.tsv'

model_config:
  name: conformer
  encoder_subsampling:
    type: conv2d
    filters: 144
    kernel_size: 3
    strides: 2
  encoder_positional_encoding: sinusoid_concat
  encoder_dmodel: 144
  encoder_num_blocks: 16
  encoder_head_size: 36
  encoder_num_heads: 4
  encoder_mha_type: relmha
  encoder_kernel_size: 32
  encoder_fc_factor: 0.5
  encoder_dropout: 0.1
  prediction_embed_dim: 320
  prediction_embed_dropout: 0
  prediction_num_rnns: 1
  prediction_rnn_units: 320
  prediction_rnn_type: lstm
  prediction_rnn_implementation: 1
  prediction_layer_norm: True
  joint_dim: 320
  joint_activation: tanh

learning_config:
  train_dataset_config:
    use_tf: True
    augmentation_config:
      after:
        time_masking:
          num_masks: 10
          mask_factor: 100
          p_upperbound: 0.05
        freq_masking:
          num_masks: 1
          mask_factor: 27
    data_paths:
      - '/content/drive/MyDrive/Colab Notebooks/TestingAudios/training.tsv'
    tfrecords_dir: /content/drive/MyDrive/Colab Notebooks/Model/TestingAudios/tfrecords
    shuffle: True
    cache: True
    buffer_size: 100
    drop_remainder: True
    stage: train

  eval_dataset_config:
    use_tf: True
    data_paths:
      - '/content/drive/MyDrive/Colab Notebooks/TestingAudios/training.tsv'
    tfrecords_dir: /content/drive/MyDrive/Colab Notebooks/Model/TestingAudios/tfrecords
    shuffle: False
    cache: True
    buffer_size: 100
    drop_remainder: True
    stage: eval

  test_dataset_config:
    use_tf: True
    data_paths:
      - '/content/drive/MyDrive/Colab Notebooks/TestingAudios/testing.tsv'
    tfrecords_dir: /content/drive/MyDrive/Colab Notebooks/Model/TestingAudios/tfrecords
    shuffle: False
    cache: True
    buffer_size: 100
    drop_remainder: True
    stage: test

  optimizer_config:
    warmup_steps: 40000
    beta1: 0.9
    beta2: 0.98
    epsilon: 1e-9

  running_config:
    batch_size: 2
    accumulation_steps: 4
    num_epochs: 10
    outdir: '/content/drive/MyDrive/Colab Notebooks/Model/'
    log_interval_steps: 300
    eval_interval_steps: 500
    save_interval_steps: 1000
    checkpoint:
      filepath: '/content/drive/MyDrive/Colab Notebooks/Model/TestingAudios/checkpoints/{epoch:02d}.h5'
      save_best_only: True
      save_weights_only: False
      save_freq: epoch
    states_dir: '/content/drive/MyDrive/Colab Notebooks/Model//TestingAudios/states'
    tensorboard:
      log_dir: '/content/drive/MyDrive/Colab Notebooks/Model/TestingAudios/tensorboard'
      histogram_freq: 1
      write_graph: True
      write_images: True
      update_freq: epoch
      profile_batch: 2

**Comment by usimarit** [comment]
@eashan2798 Can you run with beam_width = 0?

**Comment by eashan2798** [comment]
yes I executed with beam_width=0, and it computed results within a minute. Now I understand it was beam search which was taking too much time.
Thanks.
